Added Non Shibboleth WebDAV authentication and Dav Path customization to `owncloudcmd`

NvanAdrichem opened the pull request:
According to issue #4025 and @DeepDiver1975 his remark on customized dav path URLs in https://github.com/owncloud/client/issues/4007#issuecomment-152114570 I have created the following changes. I realize it currently doesn't merge, but the current Master branch gave compilation errors, hence I could not merge it by hand as I don't want to send in something untested.

DeepDiver1975 commented on src/cmd/cmd.cpp:
```diff
@@ -155,6 +157,8 @@ void help()
     std::cout << "  --password, -p [pass]  Use [pass] as password" << std::endl;
     std::cout << "  -n                     Use netrc (5) for login" << std::endl;
     std::cout << "  --non-interactive      Do not block execution with interaction" << std::endl;
+    std::cout << "  --nonshib, -ns         Use Non Shibboleth WebDAV authentication" << std::endl;
```
As soon as the dav path is customizable there is no need for the nonshib switch from my pov

NvanAdrichem replied:
Maybe not strictly necessary, but adding the nonshib switch (which is also theme customizable) is a lot easier and understandable from a user perspective than remembering and adding '--davpath remote.php/nonshib-webdav'. I don't need the davpath option, I added it while I was at it to do you a favor, I would regret missing out on the nonshib switch.

On Nov 2, 2015 23:10, Thomas Müller <notifications@github.com> wrote:

In src/cmd/cmd.cpp<https://github.com/owncloud/client/pull/4060#discussion_r43690443>:

> @@ -155,6 +157,8 @@ void help()
>      std::cout << "  --password, -p [pass]  Use [pass] as password" << std::endl;
>      std::cout << "  -n                     Use netrc (5) for login" << std::endl;
>      std::cout << "  --non-interactive      Do not block execution with interaction" << std::endl;
> +    std::cout << "  --nonshib, -ns         Use Non Shibboleth WebDAV authentication" << std::endl;


As soon as the dav path is customizable there is no need for the nonshib switch from my pov

—
Reply to this email directly or view it on GitHub<https://github.com/owncloud/client/pull/4060/files#r43690443>.

DeepDiver1975 replied:
> Maybe not strictly necessary, but adding the nonshib switch (which is also theme customizable) is a lot easier and understandable from a user perspective than remembering and adding '--davpath remote.php/nonshib-webdav'.

Since we are not talking about end-user experience I do not consider this too critical.

dragotin replied:
It is a misconception that `owncloudcmd` does not reach end users. It does.

dragotin replied:
@NvanAdrichem please do not use the "-ns" thing. Don't ask why, just don't ;-) thanks.

DeepDiver1975 replied:
end-user in the sense of lesser technical users. From a command line client user one can expect to type a url path properly as command line parameter

NvanAdrichem replied:
I consider myself an end-user... I only use the server installation of my employer, who again lease it as a service from SURFdrive, for which I am as far from an administrator, maintainer or developer as I could ever be. Being unable to make use of the owncloudcmd client through simple authentication options on non-customized, but possibly themed, servers at least bothers my end-user experience, especially since the nonshib davpath is not custom. I consider it relevant, else I would not have gone through the trouble of implementing it and filing a pull request.

DeepDiver1975 commented:
Anyway - we need an opinion from the client team @dragotin @danimo @guruz @ogoffart THX

guruz commented on src/libsync/theme.cpp:
```diff
@@ -412,5 +412,10 @@ QString Theme::webDavPath() const
 
+QString Theme::webDavPathNonShib() const
+{
+    return QLatin1String("remote.php/nonshib-webdav/");
```
This is a custom endpoint you patched into the server?
Or is that also what @DeepDiver1975 is using?

NvanAdrichem replied:
This is the default alternative path for webdav clients that do not implement Shibboleth authentication in case the server has Shibboleth Integration, hence the "nonshib". See https://doc.owncloud.org/server/9.0/admin_manual/enterprise_user_management/user_auth_shibboleth.html#webdav-support

dragotin commented:
Please resolve the merge problems against master.

Also, please update the documentation in `doc/owncloudcmd.1.rst` within this pr. 

Thanks!

NvanAdrichem commented:
Now mergeable. I removed the -ns and -dp abbreviations (I guess I was too much concerned with our recent non-results in football ;-) ).

I updated doc/owncloudcmd.1.rst and doc/owncloudcmd.rst, where one appears to be a man file and the other a section from the User Documentation. However, I noticed they are not completely in sync, and miss at least the --max-retries option that was recently added in https://github.com/owncloud/client/commit/16c078963b375fac4cb820ff9f3703c80711d3a2.

ogoffart commented:
I don't understand the nonshib dav path.
remote.php/nonshib-webdav  does not work for me on my 8.1 instance which does not have webdav.

Therefore, I personaly don't think it needs to be in the Theme.

DeepDiver1975 commented:
> remote.php/nonshib-webdav does not work for me on my 8.1 instance which does not have webdav.

you need to enable the user_shibboleth app

guruz commented:
@dragotin Can we merge this? It seems to fulfill what both @DeepDiver1975 and @NvanAdrichem need.

guruz commented:
@NvanAdrichem You did not sign the contributor agreement yet can that be? At least you don't show up here in github as contributor

https://owncloud.org/contribute/agreement/

We'd need this asap to have it in 2.1 :)

NvanAdrichem commented:
@guruz, @karlitschek has go mail!

karlitschek commented:
Thanks. confirmed! :-)

dragotin commented:
This is merged into 2.1.
